### PR TITLE
feat: revoke USDT approvals in ConfirmSwapModal

### PR DESCRIPTION
**Pull request by just-toby**
## Description

adds a new, optional, first step to the ConfirmSwapModal flow which resets USDT (mainnet) approval to zero before asking for a new, increased, approval amount. this is only needed for USDT mainnet.

from the USDT mainnet token contract:
```
 /**
    * @dev Approve the passed address to spend the specified amount of tokens on behalf of msg.sender.
    * @param _spender The address which will spend the funds.
    * @param _value The amount of tokens to be spent.
    */
    function approve(address _spender, uint _value) public onlyPayloadSize(2 * 32) {

        // To change the approve amount you first have to reduce the addresses`
        //  allowance to zero by calling `approve(_spender, 0)` if it is not
        //  already 0 to mitigate the race condition described here:
        //  https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729
        require(!((_value != 0) && (allowed[msg.sender][_spender] != 0)));

        allowed[msg.sender][_spender] = _value;
        Approval(msg.sender, _spender, _value);
    }
```

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2153/permit2-flow-special-case-usdt-approval

## Screen capture

https://github.com/Uniswap/interface/assets/66155195/df2809d7-d96b-4179-82f2-7667347587b5

### QA (ie manual testing)

- [x] verify that the step occurs when expected, and that it solves the problem of not being able to set a higher token allowance for USDT when you have a previous one set already

### Automated testing

- [x] Unit test (existing tests pass)
- [x] Integration/E2E test (added new test case in next PR)

### CODE REVIEW

**Comment by just-toby** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6726** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6726" target="_blank">Graphite</a> 
    * **PR #6727** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6727" target="_blank">Graphite</a> 
      * **PR #6728** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6728" target="_blank">Graphite</a> 
        * **PR #6729** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6729" target="_blank">Graphite</a> 
          * **PR #6730** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6730" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Uniswap/interface/6730?utm_source=stack-comment).

**Comment by just-toby** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6726** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6726" target="_blank">Graphite</a> 
    * **PR #6727** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6727" target="_blank">Graphite</a> 
      * **PR #6728** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6728" target="_blank">Graphite</a> 
        * **PR #6729** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6729" target="_blank">Graphite</a> 
          * **PR #6730** <a href="https://app.graphite.dev/github/pr/Uniswap/interface/6730" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Uniswap/interface/6730?utm_source=stack-comment).

**Comment by natew** [comment]
@just-toby so i have a swap here ready 
CleanShot 2023-06-09 at 12 53 23@2x

Hitting confirm swap goes right to asking on the wallet, how do I get it to go to the reset flow?

**Review by tinaszheng — APPROVED** [review]

File: src/components/swap/ConfirmSwapModal.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import Badge from 'components/Badge'
 import Modal, { MODAL_TRANSITION_DURATION } from 'components/Modal'
 import { RowFixed } from 'components/Row'
 import { getChainInfo } from 'constants/chainInfo'
+import { USDT } from 'constants/tokens'
```
Comment (tinaszheng) [inline src/components/swap/ConfirmSwapModal.tsx:15]: maybe rename to MAINNET_USDT? 🤔

**Review by JFrankfurt — APPROVED** [review]

**Review by cbachmeier — APPROVED** [review]
don't see a delta since last approval

**Review by JFrankfurt — APPROVED** [review]

**Pull request merged**